From 6ba616781cbbfe09d6e85d22536cccf92c9393fa Mon Sep 17 00:00:00 2001
From: Arjen Hiemstra <ahiemstra@heimr.nl>
Date: Tue, 28 Nov 2023 12:23:06 +0100
Subject: [PATCH] applets/systemmonitor: Drop "pending startup preset"

It's unclear what the purpose of this code was. At this point, it's
causing issues because the list of arguments passed to the applet
changed and we are now trying to load a preset matching the applet ID.
Removing this seems to fix things without anything else breaking.

BUG: 476666
---
 .../systemmonitor/systemmonitor.cpp           | 22 +++++--------------
 .../systemmonitor/systemmonitor.h             |  1 -
 2 files changed, 5 insertions(+), 18 deletions(-)

diff --git a/applets/systemmonitor/systemmonitor/systemmonitor.cpp b/applets/systemmonitor/systemmonitor/systemmonitor.cpp
index 0efa5c9794..a76a85bafb 100644
--- a/applets/systemmonitor/systemmonitor/systemmonitor.cpp
+++ b/applets/systemmonitor/systemmonitor/systemmonitor.cpp
@@ -25,14 +25,6 @@ SystemMonitor::SystemMonitor(QObject *parent, const KPluginMetaData &data, const
     : Plasma::Applet(parent, data, args)
 {
     setHasConfigurationInterface(true);
-
-    // Don't set the preset right now as we can't write on the config here because we don't have a Corona yet
-    if (args.count() == 2) {
-        const QString preset = args.at(1).toString();
-        if (!preset.isEmpty()) {
-            m_pendingStartupPreset = preset;
-        }
-    }
 }
 
 SystemMonitor::~SystemMonitor() = default;
@@ -47,15 +39,11 @@ void SystemMonitor::init()
     m_sensorFaceController = new KSysGuard::SensorFaceController(cg, qmlObject->engine().get());
     qmlObject->deleteLater();
 
-    if (!m_pendingStartupPreset.isNull()) {
-        m_sensorFaceController->loadPreset(m_pendingStartupPreset);
-    } else {
-        // NOTE: taking the pluginId from the child applet (cpu monitor, memory, whatever) is done implicitly by not embedding metadata in this applet
-        const QString preset = config().readEntry("CurrentPreset", pluginMetaData().pluginId());
-        // We have initialized our preset, subsequent calls should use the root-plugin id
-        config().writeEntry("CurrentPreset", "org.kde.plasma.systemmonitor");
-        m_sensorFaceController->loadPreset(preset);
-    }
+    // NOTE: taking the pluginId from the child applet (cpu monitor, memory, whatever) is done implicitly by not embedding metadata in this applet
+    const QString preset = config().readEntry("CurrentPreset", pluginMetaData().pluginId());
+    // We have initialized our preset, subsequent calls should use the root-plugin id
+    config().writeEntry("CurrentPreset", "org.kde.plasma.systemmonitor");
+    m_sensorFaceController->loadPreset(preset);
 }
 
 KSysGuard::SensorFaceController *SystemMonitor::faceController() const
diff --git a/applets/systemmonitor/systemmonitor/systemmonitor.h b/applets/systemmonitor/systemmonitor/systemmonitor.h
index 50755fe7c2..0a992ee517 100644
--- a/applets/systemmonitor/systemmonitor/systemmonitor.h
+++ b/applets/systemmonitor/systemmonitor/systemmonitor.h
@@ -47,5 +47,4 @@ public Q_SLOTS:
 
 private:
     KSysGuard::SensorFaceController *m_sensorFaceController = nullptr;
-    QString m_pendingStartupPreset;
 };
-- 
GitLab

